code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -825,6 +875,14 @@ Pass a positive integer as an argument to override a bot's default speed.
 	else	// no path, so calculate new one
 		calc_summon_path()
 
+/mob/living/simple_animal/bot/Bump(M as mob|obj) //Leave no door unopened!
```

Comment:
Doing `as X|Y|Z` is legacy code and should not be done. Doing `atom/M` would keep this in line with the parent.
```suggestion
/mob/living/simple_animal/bot/Bump(atom/M) //Leave no door unopened!
```